scripts/test.py

```diff
@@ -0,0 +1,97 @@
+
+def optimizer(optimizer_name):
+  return '--optimizer=../model_zoo/optimizers/%s.prototext' % optimizer_name
+
+def executable(cluster):
+  return '../build/%s/model_zoo/lbann' % cluster
+
+def run_lbann(model_name, reader_name, optimizer_name, output_file_name):
+  if (model_name == None or reader_name == None or optimizer_name == None or output_file_name == None):
+    raise Exception("Invalid number of arguments")
+  else:
+    os.system('hostname > %s' % output_file_name)
+    hostname_file = open(output_file_name, 'r')
+    hostname = hostname_file.readline()
+    hostname_file.close()
+    cluster = re.sub('[0-9]+\n', '.llnl.gov', hostname)
+    e = executable(cluster)
+    m = model(model_name)
+    r = reader(reader_name)
+    o = optimizer(optimizer_name)
+    # N => number of nodes                                                         
+    # p => partition
+    # t => timeout period, in minutes
+    # n => number of processes to run. MPI Rank. How many models being made.
+    # num-epochs => The number of epochs
+    os.system('salloc -N 1 -p pbatch -t 30 srun -n 2 %s %s %s %s --num_epochs=5 > %s' % (e, m, r, o, output_file_name))
+
+def get_performance(output_file_name):
+  output_file = open(output_file_name, 'r')
+  performance_dict = {}
+  for line in output_file:
+    m = re.match('(Model [0-9]+) Epoch time: ([0-9]+(\.[0-9]+)*)s; Mean minibatch time: ([0-9]+(\.[0-9]+)*)s; Min: ([0-9]+(\.[0-9]+)*)s; Max: ([0-9]+(\.[0-9]+)*)s; Stdev: ([0-9]+(\.[0-9]+)*)s', line)
+    if m:
+      model = m.group(1)
+      if model not in performance_dict.keys():
+        performance_dict[model] = {
+          'accuracies' : [],
+          'epoch_times' : [],
+          'mean_minibatch_times' : [],
+          'mins' : [],
+          'maxs' : [],
+          'stdevs' : []
+        }
+      performance_dict[model]['epoch_times'].append(float(m.group(2)))
+      performance_dict[model]['mean_minibatch_times'].append(float(m.group(4)))
+      performance_dict[model]['mins'].append(float(m.group(6)))
+      performance_dict[model]['maxs'].append(float(m.group(8)))
+      performance_dict[model]['stdevs'].append(float(m.group(10)))
+    a = re.match('(Model [0-9]+) @[0-9]+ testing steps external validation categorical accuracy: ([0-9]+(\.[0-9]+)*)', line)
+    if a:
+      model = a.group(1)
+      if model not in performance_dict.keys():
+        performance_dict[model] = {
+          'accuracies' : [],
+          'epoch_times' : [],
+          'mean_minibatch_times' : [],
+          'mins' : [],
+          'maxs' : [],
+          'stdevs' : []
+        }
+      performance_dict[model]['accuracies'].append(float(a.group(2)))
+  output_file.close()
+  return performance_dict
+
+def main(model_name, reader_name, optimizer_name, output_file_name):
+  run_lbann(model_name, reader_name, optimizer_name, output_file_name)
+  return get_performance(output_file_name)
+
+class LBANNTest(unittest.TestCase):
+  def test_performance(self):
```

Comment:
Maybe we should keep performance and accuracy together per test. However, each test will test a different model.
Perhaps rename this `test_mnist_performance` and add `test_resnet_performance` and `test_alexnet_performance`. Abstract out common code.